e/tool/tele/cli/helm.go

```diff
@@ -61,12 +61,22 @@ func (o *repoAddOptions) repoAdd() error {
 	lockCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 	locked, err := fileLock.TryLockContext(lockCtx, time.Second)
-	if err == nil && locked {
-		defer fileLock.Unlock()
-	}
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	if !locked {
+		return trace.LimitExceeded("unable to obtain file lock")
+	}
```

Comment:
Yes, I've looked at the helm's code (this is a copy/paste from the helm repository) and the [flock](https://github.com/gofrs/flock) repository and the only way the code would get (`false`, `nil`) is handled inside the flock library itself ([here](https://github.com/gofrs/flock/blob/75ec202958fd1654ff47064ba3ebffe14443f391/flock_unix.go#L155) and [here](https://github.com/gofrs/flock/blob/75ec202958fd1654ff47064ba3ebffe14443f391/flock.go#L105-L113)) so it never escapes to the caller although the implementation is as if it could, in which case my changes are more robust in preventing concurrent access.